src/main/java/org/cryptomator/windows/autostart/WinShellLinks.java

```diff
@@ -0,0 +1,43 @@
+package org.cryptomator.windows.autostart;
+
+import org.cryptomator.windows.common.NativeLibLoader;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Interface to the native Windows shell link interface.
+ * <p>
+ * For more info, see https://docs.microsoft.com/de-de/windows/win32/shell/links.
+ */
+public class WinShellLinks {
+
+	/**
+	 * Create a Windows shortcut file to the given target, at the specified location and with a description.
+	 *
+	 * @param target path of the target the shortcut points to
+	 * @param storagePath path where the shortcut is created
+	 * @param description string inserted in the description field of the shortcut.
+	 * @return {@code 0} if everything worked, otherwise an HRESULT error code
+	 */
+	public int createShortcut(String target, String storagePath, String description) {
+		return Native.INSTANCE.createShortcut(
+				nullterminateAndConvert(target),
+				nullterminateAndConvert(storagePath),
+				nullterminateAndConvert(description)
+		);
+	}
+
+	private byte[] nullterminateAndConvert(String source) {
+		return (source + "\0").getBytes(StandardCharsets.UTF_16LE);
```

Comment:
should rather be:

```suggestion
		byte[] bytes = source.getBytes(StandardCharsets.UTF_16LE)
		return Arrays.copyOf(bytes, bytes.length + 1); // add padding
```

After all, we want one null-byte at the end of the array, not the UTF16-representation of a null-character.

Reply:
🙈 This was my first thought, but somehow I convinced myself it needs to be different